src/Analysis/Engine/Impl/AnalysisUnit.cs

```diff
@@ -342,6 +344,22 @@ internal AnalysisUnit(ScopeStatement ast, IScope scope)
         internal virtual ILocationResolver AlternateResolver => null;
 
         ILocationResolver ILocationResolver.GetAlternateResolver() => AlternateResolver;
+
+        readonly Lazy<AnalysisUnit> _externalAnnotationAnalysisUnit;
+        protected internal virtual AnalysisUnit GetExternalAnnotationAnalysisUnit() {
+            if (Ast != Tree)
+                return null;
+
+            return _externalAnnotationAnalysisUnit.Value;
+        }
+
+        AnalysisUnit GetExternalAnalysisUnitImpl() {
+            string analysisModuleName = ProjectEntry.ModuleName + PythonAnalyzer.AnnotationsModuleSuffix;
+            if (!ProjectEntry.ProjectState.Modules.TryImport(analysisModuleName, out var analysisModuleReference))
+                return null;
+
+            return analysisModuleReference.AnalysisModule?.AnalysisUnit;
```

Comment:
Imported non-user files only get through AST analysis, do you actually get analysis unit here?

Reply:
I am not sure I understand this comment.

PythonAnalyzer.AddModuleAnnotations adds annotations module to its Modules section. Just like for normal modules, you'd have to call .Analyze on the IPythonProjectEntry that is returned by AddModuleAnnotations.

Is there some other kind of analysis, that I am missing?